### ISSUE TITLE
[Slim] How to make sure that multiple clones are seeing different images while training inception on 4 GPUs.

### ISSUE THREAD

**Issue by SnShine** [tensorflow/models#2903]
I'm using `num_clones=1`, `batch_size=96` on a P3 instance to train inception net with custom images. It's taking `0.5secs/step` and as I set learning_rate decay after 2 epochs, learning_rate **decays after 280 steps** which is correct with my calculations.

Now, when using 4 GPUs on the same instance with `num_clones=4`, `batch_size=384`, I get an error saying it can't load them into GPU. So, I'm assuming we need to specify batch_size for each clone but not for all clones combined.

So, when I put `num_clones=4`, `batch_size=96` with appropriate num_worker and stuff, my 4 GPUs are running at full volatile and it takes `0.6secs/step` which is reasonable, I guess.

So, if we are crunching 4 times the number of images in each step, my learning rate **should decay after 280/4=70 steps.** But that's not happening! 😨 Learning rate is decaying only at 280 steps. Does it mean all 4 GPUs are analyzing the same set of 96 images every time?

**Comment by tatatodd** [comment]
This question is better asked on  [StackOverflow](http://stackoverflow.com/questions/tagged/tensorflow) since it is not a  bug or feature request. There is also a larger community that reads questions there. Thanks!

**Comment by DavidWiesner** [comment]
I also think this is a bug in the implementation in `train_image_classifier.py`. There is also a [comment in this thread](https://github.com/tensorflow/models/issues/1428#issuecomment-322732761) mention this problem. The calculation of decay steps in  [`train_image_classifier.py#L239`](https://github.com/tensorflow/models/blob/master/research/slim/train_image_classifier.py#L239) do reflect the numbers of replicas but not the number of clones.

**Comment by SnShine** [comment]
I'm not so sure. As a work around, users can modify `epoches_per_decay` by dividing it with your number of clones to follow the same decay pattern when using one clone.
